hazelcast/src/main/java/com/hazelcast/config/MultiMapConfig.java

```diff
@@ -409,29 +419,47 @@ public boolean equals(Object o) {
         if (statisticsEnabled != that.statisticsEnabled) {
             return false;
         }
-        if (!name.equals(that.name)) {
+        if (name != null ? !name.equals(that.name) : that.name != null) {
             return false;
         }
-        if (quorumName != null ? !quorumName.equals(that.quorumName) : that.quorumName != null) {
+        if (valueCollectionType != null ? !valueCollectionType.equals(that.valueCollectionType) :
+                that.valueCollectionType != null) {
             return false;
         }
-        if (valueCollectionType != null
-                ? !valueCollectionType.equals(that.valueCollectionType) : that.valueCollectionType != null) {
+        if (listenerConfigs != null ? !listenerConfigs.equals(that.listenerConfigs) : that.listenerConfigs != null) {
             return false;
         }
-        return listenerConfigs != null ? listenerConfigs.equals(that.listenerConfigs) : that.listenerConfigs == null;
+        if (quorumName != null ? !quorumName.equals(that.quorumName) : that.quorumName != null) {
+            return false;
+        }
+        return mergePolicyConfig != null ? mergePolicyConfig.equals(that.mergePolicyConfig) : that.mergePolicyConfig == null;
     }
 
     @Override
-    public int hashCode() {
-        int result = name.hashCode();
+    @SuppressWarnings("checkstyle:npathcomplexity")
+    public final int hashCode() {
```

Comment:
why is this method final?

Reply:
It's due to the used testing library for `equals()` and `hashCode()` methods, which was introduced by Vassilis with the dynamic config feature. Most of these methods in the config classes are `final` due to this.

Reply:
Actually it doesn't seem a good practice to me making public methods final for testing purposes.

Reply:
The author of that test library seems to think that it's a good practise to have `equals()` and `hashCode()` final. So it's not an artificial requirement, they actively check for this: http://jqno.nl/equalsverifier/errormessages/subclass-equals-is-not-final/

If you think this is a problem, please open an issue and get Vassilis involved as well. As I said, this was introduced with the dynamic config feature (and it actually resolved the issue of not testing `equals()` and `hashCode()` properly or at all).

Reply:
Here is an explanation from the author on this: https://github.com/jqno/equalsverifier/blob/master/src/main/java/nl/jqno/equalsverifier/Warning.java#L169